packages/language-support/src/syntaxValidation/syntaxValidation.ts

```diff
@@ -138,6 +142,60 @@ function generateProcedureNotFoundError(
   return error;
 }
 
+function generateProcedureDeprecatedWarning(
+  parsedProcedure: ParsedProcedure,
+): SyntaxDiagnostic {
+  const rawText = parsedProcedure.rawText;
+  const nameChunks = rawText.split('\n');
+  const linesOffset = nameChunks.length - 1;
+  const lineIndex = parsedProcedure.line - 1;
+  const startColumn = parsedProcedure.column;
+  const endColumn =
+    linesOffset == 0
+      ? startColumn + rawText.length
+      : nameChunks.at(-1)?.length ?? 0;
```

Comment:
This piece of code is repeated several times in the codebase already, so I'd suggest maybe splitting it to a helper method?